File: leetcode/two_sum_ii.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if len(numbers) < 2:
            return []

        l, r = 0, len(numbers) - 1

        while l < r:
            s = numbers[l] + numbers[r]
            if s == target:
                return [l + 1, r + 1]
            if s < target:
                l += 1
            else:
                r -= 1
        return []

File: leetcode/two_sum_ii.go
package leetCode

func twoSum(numbers []int, target int) []int {
    l, r := 0, (len(numbers) - 1)
    for l < r {
        s := numbers[l] + numbers[r]
        if s == target {
            return []int{l + 1, r + 1}
        }
        if s < target {
            l++
        } else {
            r--
        }
    }
    return []int{0, 0}
}
